spec/colorizer_spec.rb

```diff
@@ -1,19 +1,25 @@
 require './lib/colorizer'
 
 describe 'colorizer' do
-  let(:colorizer) { Colorizer.new }
+  let(:colorizer_unsupported) { Colorizer.new(1) }
 
-  let(:test_string) { 'test' }
+  let(:colorizer_supported) { Colorizer.new(8) }
 
-  it 'colors a string magenta' do
-    expect(colorizer.magenta(test_string)).to eq "\e[35m#{test_string}\e[0m"
+  let(:test) { 'test' }
+
+  it 'only colors a string if supported' do
+    expect(colorizer_unsupported.position(test)).to eq test
+  end
+
+  it 'colors symbol_1 if supported' do
+    expect(colorizer_supported.symbol_1(test)).to eq "\e[34m#{test}\e[0m"
   end
 
-  it 'colors a string yellow' do
-    expect(colorizer.yellow(test_string)).to eq "\e[33m#{test_string}\e[0m"
+  it 'colors symbol_2 if supported' do
+    expect(colorizer_supported.symbol_2(test)).to eq "\e[36m#{test}\e[0m"
   end
 
-  it 'colors a string cyan' do
-    expect(colorizer.cyan(test_string)).to eq "\e[36m#{test_string}\e[0m"
+  it 'colors a position if supported' do
+    expect(colorizer_supported.position(test)).to eq "\e[33m#{test}\e[0m"
```

Comment:
I might opt for using a context to differentiate the specs for supported and unsupported color. You could move the relevant lets to those contexts as well.